internal/provider/kubernetes/controller.go

```diff
@@ -1489,118 +1556,143 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 		}
 	}
 
-	// Watch ClientTrafficPolicy
-	ctpPredicates := []predicate.TypedPredicate[*egv1a1.ClientTrafficPolicy]{
-		predicate.TypedGenerationChangedPredicate[*egv1a1.ClientTrafficPolicy]{},
-	}
-	if r.namespaceLabel != nil {
-		ctpPredicates = append(ctpPredicates, predicate.NewTypedPredicateFuncs[*egv1a1.ClientTrafficPolicy](func(ctp *egv1a1.ClientTrafficPolicy) bool {
-			return r.hasMatchingNamespaceLabels(ctp)
-		}))
-	}
+	r.ctpCRDExists = r.crdExists(mgr, resource.KindClientTrafficPolicy, egv1a1.GroupVersion.String())
+	if !r.ctpCRDExists {
+		r.log.Info("ClientTrafficPolicy CRD not found, skipping ClientTrafficPolicy watch")
+	} else {
+		// Watch ClientTrafficPolicy
+		ctpPredicates := []predicate.TypedPredicate[*egv1a1.ClientTrafficPolicy]{
+			predicate.TypedGenerationChangedPredicate[*egv1a1.ClientTrafficPolicy]{},
+		}
+		if r.namespaceLabel != nil {
+			ctpPredicates = append(ctpPredicates, predicate.NewTypedPredicateFuncs[*egv1a1.ClientTrafficPolicy](func(ctp *egv1a1.ClientTrafficPolicy) bool {
+				return r.hasMatchingNamespaceLabels(ctp)
+			}))
+		}
 
-	if err := c.Watch(
-		source.Kind(mgr.GetCache(), &egv1a1.ClientTrafficPolicy{},
-			handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, ctp *egv1a1.ClientTrafficPolicy) []reconcile.Request {
-				return r.enqueueClass(ctx, ctp)
-			}),
-			ctpPredicates...)); err != nil {
-		return err
-	}
+		if err := c.Watch(
+			source.Kind(mgr.GetCache(), &egv1a1.ClientTrafficPolicy{},
+				handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, ctp *egv1a1.ClientTrafficPolicy) []reconcile.Request {
+					return r.enqueueClass(ctx, ctp)
+				}),
+				ctpPredicates...)); err != nil {
+			return err
+		}
 
-	if err := addCtpIndexers(ctx, mgr); err != nil {
-		return err
+		if err := addCtpIndexers(ctx, mgr); err != nil {
+			return err
+		}
 	}
 
-	// Watch BackendTrafficPolicy
-	btpPredicates := []predicate.TypedPredicate[*egv1a1.BackendTrafficPolicy]{
-		predicate.TypedGenerationChangedPredicate[*egv1a1.BackendTrafficPolicy]{},
-	}
-	if r.namespaceLabel != nil {
-		btpPredicates = append(btpPredicates, predicate.NewTypedPredicateFuncs[*egv1a1.BackendTrafficPolicy](func(btp *egv1a1.BackendTrafficPolicy) bool {
-			return r.hasMatchingNamespaceLabels(btp)
-		}))
-	}
+	r.btpCRDExists = r.crdExists(mgr, resource.KindBackendTrafficPolicy, egv1a1.GroupVersion.String())
+	if !r.btpCRDExists {
+		r.log.Info("BackendTrafficPolicy CRD not found, skipping BackendTrafficPolicy watch")
```

Comment:
I think there're two approaches:
* Adds an opt-out configuration in `EnvoyGateway` API for EG CRs
* Reports a warning in the log

This doesn't have to be handled in this PR as we're approaching v1.2.0. It can be tracked in an issue and addressed later.

Reply:
raised https://github.com/envoyproxy/gateway/issues/4553